diff -uNr a/drivers/net/ethernet/stmicro/stmmac/dwmac-stm32.c b/drivers/net/ethernet/stmicro/stmmac/dwmac-stm32.c
--- a/drivers/net/ethernet/stmicro/stmmac/dwmac-stm32.c	2025-04-22 02:36:47.860919988 +0000
+++ b/drivers/net/ethernet/stmicro/stmmac/dwmac-stm32.c	2025-04-22 05:01:48.050702543 +0000
@@ -8,6 +8,7 @@
 
 #include <linux/clk.h>
 #include <linux/kernel.h>
+#include <linux/gpio/consumer.h>
 #include <linux/mfd/syscon.h>
 #include <linux/module.h>
 #include <linux/of.h>
@@ -103,6 +104,7 @@
 	u32 speed;
 	const struct stm32_ops *ops;
 	struct device *dev;
+	struct gpio_desc *phy_reset;
 };
 
 struct stm32_syscfg_pmcsetr {
@@ -482,6 +484,21 @@
 	return 0;
 }
 
+
+static int stm32_dwmac_phy_reset(struct device *dev, struct stm32_dwmac *dwmac)
+{
+
+       dwmac->phy_reset = devm_gpiod_get_optional(dev, "st,phy-reset", GPIOD_OUT_LOW);
+       if (IS_ERR(dwmac->phy_reset))
+          return PTR_ERR(dwmac->phy_reset);
+       if (dwmac->phy_reset) {
+                usleep_range(100, 200);
+               gpiod_set_value_cansleep(dwmac->phy_reset, 1);
+       }
+       return 0;
+}
+
+
 static int stm32_dwmac_probe(struct platform_device *pdev)
 {
 	struct plat_stmmacenet_data *plat_dat;
@@ -490,6 +507,15 @@
 	const struct stm32_ops *data;
 	int ret;
 
+	dwmac = devm_kzalloc(&pdev->dev, sizeof(*dwmac), GFP_KERNEL);
+	if (!dwmac)
+			return -ENOMEM;
+
+	/* Optional Phy reset */
+	ret = stm32_dwmac_phy_reset(&pdev->dev, dwmac);
+	if (ret)
+		return ret;
+
 	ret = stmmac_get_platform_resources(pdev, &stmmac_res);
 	if (ret)
 		return ret;
